### PR TITLE
unit-tests-jest - Tim Silva

### PR DESCRIPTION
**Pull request by silva-tim**

### CODE REVIEW

File: unit-tests-jest/numbers.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * `toDollars(11.341)` returns `$11.34`.
+ */
+export function toDollars(amount) {
+  if (Number.isInteger(amount)) {
+    return `$${amount}.00`;
+  } else {
+    const num = `${Math.floor(amount * 100)}`;
+    const dec = num.slice(-2);
+    return `$${num.slice(0, -2) + '.' + dec}`;
+  }
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:21]: The easiest way to do this is to use `amount.toFixed(2)`.

File: unit-tests-jest/numbers.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+/**
+ * Returns a new array of numbers where every entry has been divided by the
+ * given divisor. Does not modify the original array.
+ * @param {Array} numbers The array of numbers to be divided by the divisor.
+ * @param {number} divisor The number to divide into each element of `numbers`.
+ * @returns a new array.
+ */
+export function divideBy(numbers, divisor) {
+  for (let i = 0; i < numbers.length; i++) {
+    numbers[i] = numbers[i] / divisor;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:33]: This modifies the original array but the spec says "Does not modify the original array."

File: unit-tests-jest/numbers.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+/**
+ * Modifies an object by multiplying the value of each key, but only if that
+ * value is a number. Key names are arbitrary.
+ * @param {Object} obj The object to modify.
+ * @param {number} multiplier The number to multiply by.
+ * @returns the input object.
+ */
+export function multiplyBy(obj, multiplier) {
+  const result = { ...obj };
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:46]: This copies the original object but the spec says "Modifies an object".

**Review by RobertGardner — CHANGES_REQUESTED** [review]
You got the important part of the exercise, but since you're finished so much earlier than everyone else, I think you can work on the two things that don't match spec. :-)

File: unit-tests-jest/numbers.js (hunk 3 above)
Comment (silva-tim) [reply unit-tests-jest/numbers.js]: Fixed

File: unit-tests-jest/numbers.js [hunk 4]
```diff
@@ -15,9 +15,7 @@ export function toDollars(amount) {
   if (Number.isInteger(amount)) {
     return `$${amount}.00`;
   } else {
-    const num = `${Math.floor(amount * 100)}`;
-    const dec = num.slice(-2);
-    return `$${num.slice(0, -2) + '.' + dec}`;
+    return `$${amount.toFixed(2)}`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:18]: `toFixed` gives the correct result even if it's an integer.

File: unit-tests-jest/numbers.js [hunk 5]
```diff
@@ -29,10 +27,7 @@ export function toDollars(amount) {
  * @returns a new array.
  */
 export function divideBy(numbers, divisor) {
-  for (let i = 0; i < numbers.length; i++) {
-    numbers[i] = numbers[i] / divisor;
-  }
-  return numbers;
+  return numbers.map((i) => i / divisor);
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:30]: Did you think about how to test that it does not modify the array?

File: unit-tests-jest/numbers.js [hunk 6]
```diff
@@ -43,11 +38,10 @@ export function divideBy(numbers, divisor) {
...
-  for (const key in result) {
-    if (typeof result[key] === 'number') {
-      result[key] = result[key] * multiplier;
+  for (const key in obj) {
+    if (typeof obj[key] === 'number') {
+      obj[key] = obj[key] * multiplier;
     }
   }
-  return result;
+  return obj;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js:46]: Did you think about how to test that it does modify the object?

**Review by RobertGardner — APPROVED** [review]

**Pull request merged**